cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -74,10 +74,9 @@ struct EmitInfo {
   std::shared_ptr<Schema> schema;
 };
 
-template <typename RelMessage>
-Result<EmitInfo> GetEmitInfo(const RelMessage& rel,
+Result<EmitInfo> GetEmitInfo(const substrait::RelCommon& rel_common,
```

Comment:
Why this change

Reply:
This is a refactoring that created two overloads of the functions, the original first accepting a message (of some type, using the template) and the new second accepting a `RelCommon`. This is possible because the implementation of the original function is mostly just looking at the `RelCommon` of the message. This allows `MakeAggregateDeclaration` to be a non-template function by [invoking a non-template `ProcessEmit`](https://github.com/apache/arrow/pull/34627/files#diff-2c7a2e65948433ab4aa1fb595d3132338b0e9684ed57f8021c28e448dd8a8ec2R384).